apps/rush-lib/src/logic/DependencyIntegrator.ts

```diff
@@ -0,0 +1,355 @@
+                project: this._rushConfiguration.getProjectByName(consumer)!,
+                packageName: packageName,
+                newVersion: version
+              });
+            }
+          }
+        }
+      }
+    }
+
+    this.updateProjects(packageUpdates);
+
+    if (skipUpdate) {
+      return Promise.resolve();
+    }
+
+    const purgeManager: PurgeManager = new PurgeManager(this._rushConfiguration);
+    const installManager: InstallManager = new InstallManager(this._rushConfiguration, purgeManager);
+    const installManagerOptions: IInstallManagerOptions = {
+      debug: debugInstall,
+      allowShrinkwrapUpdates: true,
+      bypassPolicy: false,
+      noLink: false,
+      fullUpgrade: false,
+      recheckShrinkwrap: false,
+      networkConcurrency: undefined,
+      collectLogFile: true
+    };
+
+    return installManager.doInstall(installManagerOptions)
+      .then(() => {
+        purgeManager.deleteAll();
+      })
+      .catch((error) => {
+        purgeManager.deleteAll();
+        throw error;
+      });
+  }
+
+  /**
+   * Updates several projects' package.json files
+   */
+  public updateProjects(projectUpdates: Array<IUpdateProjectOptions>): void {
+    for (const update of projectUpdates) {
+      this.updateProject(update);
+    }
+  }
+
+  /**
+   * Updates a single project's package.json file
+   */
+  public updateProject(options: IUpdateProjectOptions): void {
+    let { dependencyKind } = options;
+    const {
+      project,
+      packageName,
+      newVersion,
+      doNotSave
+    } = options;
+    const packageJson: IPackageJson = project.packageJson;
+
+    let oldDependencyKind: DependencyKind | undefined = undefined;
+    if (packageJson.dependencies && packageJson.dependencies[packageName]) {
+      oldDependencyKind = DependencyKind.Dependency;
+    } else if (packageJson.devDependencies && packageJson.devDependencies[packageName]) {
+      oldDependencyKind = DependencyKind.DevDependency;
+    }
+
+    if (!dependencyKind && !oldDependencyKind) {
+      throw new Error(`Cannot auto-detect dependency type of "${packageName}" for project "${project.packageName}"`);
+    }
+
+    if (!dependencyKind) {
+      dependencyKind = oldDependencyKind;
+    }
+
+    // update the dependency
+    if (dependencyKind === DependencyKind.Dependency) {
+      packageJson.dependencies = this._updateDependency(packageJson.dependencies, packageName, newVersion);
+    } else if (dependencyKind === DependencyKind.DevDependency) {
+      packageJson.devDependencies = this._updateDependency(packageJson.devDependencies, packageName, newVersion);
+    }
+
+    if (!doNotSave) {
+      // overwrite existing file
+      const packageJsonPath: string
+        = path.join(project.projectFolder, FileConstants.PackageJson);
+      JsonFile.save(project.packageJson, packageJsonPath);
+
+      console.log(colors.green('Wrote ') + packageJsonPath);
+    }
+  }
+
+  private _getNormalizedVersionSpec(
```

Comment:
>_getNormalizedVersionSpec [](start = 10, length = 25)

Add some docs on what these parameters are -- I know it's documented above, but it would be super useful to have a copy here when reading this complex logic